pkg/components/aws-ebs-csi-driver/component.go

```diff
@@ -76,8 +76,8 @@ func (c *component) RenderManifests() (map[string]string, error) {
 	return renderedFiles, nil
 }
 
-func (c *component) Metadata() components.Metadata {
-	return components.Metadata{
+func (c *component) Metadata() *components.Metadata {
```

Comment:
I wouldn't make it a pointer. The metadata can return the values from the config (though this is still a slight abuse IMO), but generally metadata should be immutable and easily accessible. If we make it a pointer, we should check everywhere, that it is not nil.